### DISCUSSION TITLE
Watts -> Running

### DISCUSSION THREAD

**Discussion by kenlpeters** [philosowaffle/peloton-to-garmin#85] [Q&A] [answered]
I actually have 2 questions:

Background:  I have the Tread + and it calculates Watts and they are in the TCX file as activityExtension:Watts.  Currently I am using an Echelon and Watts do not come into Peloton from that Bike.  But my Peloton Bike comes in 2 weeks from now.  So...

1)  Does Garmin register Watts for running activities and is that why Watts are not being imported?  
2)  Will Watts come for Bike rides?

**Comment by philosowaffle** [comment, marked as answer]
1. Garmin does register and display Watts for running activities, atleast it does for my Stryd footpod. But Stryd records FIT files which are much more feature rich than TCX files. If you can send me your TCX that has running power in it, when I have some free time I can investigate what it might take to get running power into Garmin.
2. Watts do currently display for Bike rides. Here is an example of what the bike data looks like in Garmin.

<img width="1699" alt="Screen Shot 2021-03-31 at 8 34 51 AM" src="https://user-images.githubusercontent.com/7275041/113152727-049b3a00-91fc-11eb-9a76-e4da46005c3b.png">

**Reply by kenlpeters** [reply]
Awesome - I happy about the bikes. That is what I care about more.  i would great to figure out to how to get the Watts in runs.

**Reply by kenlpeters** [reply]
https://github.com/kenlpeters/gymscheduler/blob/b930b3f006278f182176246861bdbdbe8f10cf01/1617096967-20%20min%20Hike%20with%20Selena%20Samuela-79b059f326304b639e706d32f5f0559d.tcx

**Reply by kenlpeters** [reply]
I noticed if I replaced the activityExtension with ns3 it imports.  I added code to my fork that replaced activityExtension to ns3 everything imported correctly.

**Reply by philosowaffle** [reply]
Can you submit a pr with this change?

**Reply by kenlpeters** [reply]
Not sure I did it correct but I tried.  There is pull request with my changes in it.

**Reply by philosowaffle** [reply]
So I tested uploading that TCX file to my account (using the original `activityExtension` namespace) and watts displayed for me without any modification.  So I think the root cause may be something else. 

I wonder if the solution was really related to setting the device and author info?  I know that sometimes, Garmin only displays certain features if it detects your device supports those features.

![2021-04-03_11-21-23](https://user-images.githubusercontent.com/7275041/113484673-30722600-946f-11eb-8967-c75aba89bf41.png)

**Reply by kenlpeters** [reply]
Interesting.  thanks for letting me know.
